### DISCUSSION TITLE
onMissingChildren

### DISCUSSION THREAD

**Discussion by ChristopherMeek** [lukasbach/react-complex-tree#142] [Ideas]
Hey All,

I'm trying to put together a PR to complete the onMissingChildren functionality.  Modifying TreeItem to do a check similar to the missing item check is straightforward enough, something like ...

```
if(item.hasChildren && isExpanded && item.children === undefined && !hasRequestedChildren) {
  setHasRequestedChildren(true);
  environment.onMissingChildren?.([props.itemIndex]);
}
```

This works fine for controlled environments.  For uncontrolled environments a bit more work needs to be done to the dataprovider and UncontrolledEnvironment.  I'm thinking of adding `getChildren?: (itemId: TreeItemIndex) => Promise<TreeItemIndex[]>;` to the `TreeDataProvider` interface and then an `onMissingChildren` handler to `UncontrolledEnvironment` that pushes the received ids into item in much the same way as `onMissingItems`.  

Of course I will add tests, test-data and stories to complete the PR.

Does this sound like a reasonable approach?

**Comment by youcandanch** [comment]
Time to dredge up some zombie discussions!

@lukasbach any thoughts around this one? I think what @ChristopherMeek is proposing seems reasonable, and I'm happy to take a stab at it, but wasn't 100% sure what you were thinking as far as supporting this prop.

Edit: Ended up going in a different direction here that works for me, but would still be happy to help if this is something desired in the lib!
